draft-ietf-idr-5g-edge-service-metadata-24.xml

```diff
@@ -340,37 +354,59 @@
-          Sub-TLVs present in the BGP Metadata Path attribute is determined by
-          the configuration. When a BGP Speaker does not recognize some of the
-          Sub-TLVs within one Metadata Path Attribute in a BGP UPDATE message,
-          the BGP Speaker should forward the received BGP UPDATE message
-          without any change if the transitive bit is set to 1 <xref
-          target="RFC4271"/>. The domain ingress nodes should process the
+          Sub-TLVs present in the BGP Metadata Path Attribute is determined by
+          configuration.  The domain ingress nodes should process the
           recognized Sub-TLVs carried by the Metadata Path Attribute and
           ignore the unrecognized Sub-TLVs. By default, a BGP speaker does not
           report any unrecognized Sub-TLVs within a Metadata Path Attribute
           unless configured to send a notification to its management system.
           The ingress node should be configured with an algorithm to combine
           the recognized metrics carried by the Sub-TLVs within a Metadata
           Path Attribute of the received BGP UPDATE message.</t>
+	  <!-- XXX JMH "combine" is odd.  Perhaps utilize? -->
+
+	  <!-- XXX JMH - the text below doesn't make much bgp sense.  Path
+	  attributes apply to all of the NLRI carried in the update message.  I
+	  recommend striking this section and have commented it out to flag
+	  doing so.
 
           <t>The metrics Sub-TLVs included in the Metadata Path Attribute
-          apply to all the address families carried in the NLRI field of the
+          apply to all of the address families carried in the NLRI field of the
           BGP UPDATE message <xref target="RFC4271"/>. For a multi-protocol
           BGP UPDATE message <xref target="RFC4760"/> <xref
           target="RFC7606"/>, the metrics Sub-TLVs included in the Metadata
           Path Attribute apply to all the AFIs/SAFIs address families carried
           by the MP_REACH_NLRI.</t>
+	  XXX JMH -->
         </section>
       </section>
 
       <section title="Metadata Path Attribute TLV">
+	<!-- XXX JMH - this section is problematic.  While I could certainly
+	clean it up to address this comment, the presence of the "reserved"
+	field is worrying.
+
+	In general, we don't document the attr flags, or attribute in BGP
+	packet diagrams.
+	The motivation is that we don't want to accidentally redefine the
+	semantics of the core protocol field.
+
+	Similarly, the length field is NOT guaranteed to be one octet and that
+	depends on the implementation's encoding of the field.  Generally, the
+	implementation should use a one octet length field if the data is <=
+	255 octets in length.  However, implementations MUST be able to decode
+	a two octet length that contains data <= 255 in length.
+
+	If you want a RESERVED field in the path attribute, it's the first
+	octet of your data.  Is that what is really wanted here? -->
```

Comment:
How about replacing with the following paragraph? 

_"The Metadata Path Attribute is an optional non-transitive BGP path attribute. IANA is expected to assign a value as the type code of the attribute in the "BGP Path Attributes" registry. 
The attribute is composed of a set of Type-Length-Value (TLV) encodings.
Each TLV contains information corresponding to a particular metadata type. A Metadata TLV is structured as shown below:"_

Reply:
We can look at that, but need to resolve the reserved field.  See prior point above.